### PR TITLE
Feature/bugzilla bug 60628 ant get task to accept arbitrary header

### PR DESCRIPTION
**Pull request by arcadius**
[AA] bugzilla Bug 60628 Ant-Get-Task-To-Accept-Arbitrary-Header

### CODE REVIEW

**Comment by arcadius** [comment]
@mbenson  , would you be able to review this pull request for me?
It is very similar to the email header that you worked on.

**Comment by janmaterne** [comment]
Haven't done a deep check, but first comments:
- overall it seems to be ok
- get.html: maybe a hint that key and value are trimmed and must not be null or empty string
- get.html, example: value has doubled equals sign
- get.xml/GetTest.java: method name doesn't match target name
- Get.java: nice to reuse email.Header
- Get.java: trimToNull() could be moved to org.apache.tools.ant.util.StringUtils
- Get.java:764: the if(isEmpty) is not required

@mbenson: Are you reviewing this?

File: manual/Tasks/get.html [hunk 1]
```diff
@@ -170,6 +170,29 @@
...
+
+<table width="60%" border="1" cellpadding="2" cellspacing="0">
+  <tr>
+    <td valign="top"><b>Attribute</b></td>
+    <td valign="top"><b>Description</b></td>
+    <td align="center" valign="top"><b>Required</b></td>
+  </tr>
+  <tr>
+    <td valign="top">name</td>
+    <td valign="top">The name or key of this header.</td>
```
Comment (arcadius) [inline manual/Tasks/get.html]: Added comment about empty space being trimmed

File: manual/Tasks/get.html [hunk 2]
```diff
@@ -234,6 +257,17 @@
   &lt;url url=&quot;http://ant.apache.org/faq.html&quot;/&gt;
 &lt;/get&gt;
 </pre>
+
+<p>With custom HTTP headers</p>
+<pre>
+&lt;get src=&quot;http://ant.apache.org/index.html&quot; dest=&quot;downloads&quot;&gt;
+  &lt;header name=&quot;header1&quot; value==&quot;headerValue1&quot; /&gt;
```
Comment (arcadius) [inline manual/Tasks/get.html]: fixed typo ```==```

File: src/main/org/apache/tools/ant/taskdefs/Get.java [hunk 3]
```diff
@@ -726,6 +761,14 @@ private URLConnection openConnection(final URL aSource) throws IOException {
                 connection.setRequestProperty("Accept-Encoding", GZIP_CONTENT_ENCODING);
             }
 
+            if (!headers.isEmpty()) {
```
Comment (arcadius) [inline src/main/org/apache/tools/ant/taskdefs/Get.java]: Yes, agree, ```headers``` empty check is not necessary +1

File: src/main/org/apache/tools/ant/taskdefs/Get.java [hunk 4]
```diff
@@ -483,6 +489,35 @@ public void setTryGzipEncoding(boolean b) {
...
+        if (header != null) {
+            String key = trimToNull(header.getName());
+            String value = trimToNull(header.getValue());
+            if (key != null && value != null) {
+                this.headers.put(key, value);
+            }
+        }
+    }
+
+    private String trimToNull(String inputString) {
```
Comment (arcadius) [inline src/main/org/apache/tools/ant/taskdefs/Get.java]: moved this to StringUtils

File: src/etc/testcases/taskdefs/get.xml [hunk 5]
```diff
@@ -98,6 +98,34 @@
...
   </target>
 
+  <target name="testTwoHeaders">
+    <get src="http://www.apache.org/" dest="get.tmp">
+      <header name="header1" value="header1Value"/>
+      <header name="header2" value="header2Value"/>
+    </get>
+  </target>
+
+  <target name="testEmptyHeaders">
```
Comment (arcadius) [inline src/etc/testcases/taskdefs/get.xml]: changed all these names to match the one in the ```GetTest```

**Comment by arcadius** [comment]
Thank you very much @janmaterne for taking the time to review the code.
Most appreciated. :+1: 

I have made all suggested changes.

Feel free to have a look.

Many thanks.

**Comment by arcadius** [comment]
Hello @janmaterne ... is there anything i can do to help get this through?

Thank you very much.

Arcadius.

**Comment by janmaterne** [comment]
committed to master branch (Ant 1.10.x).
thanks for the patch

**Comment by arcadius** [comment]
Thank you very much @janmaterne for your help

**Pull request merged**